Bio/Align/__init__.py

```diff
@@ -1107,6 +1110,169 @@ def parse_printed_alignment(cls, lines):
+
+            The pairwise alignment would look like
+            >>> print(f"Reference: {pwa[0]}")
+            Reference: ACGT
+            >>> print(f"Seq1:      {pwa[1]}")
+            Seq1:      AC-T
+
+            The three sequence alignment would look like
+            >>> str(not_pwa[0])
+            'ACG-T'
+            >>> str(not_pwa[1])
+            'ACGGT'
+            >>> str(not_pwa[2])
+            'A---T'
+
+            Now, we can combine these alignments into a multiple sequence alignment:
+            >>> msa = Alignment.from_alignments_with_same_reference([pwa, not_pwa])
+            >>> str(msa[0])
+            'ACG-T'
+            >>> str(msa[1])
+            'AC--T'
+            >>> str(msa[2])
+            'ACGGT'
+            >>> str(msa[3])
+            'A---T'
+
+        Example 2: Using SeqRecord Objects with Metadata
+            Consider the following reference and sequences with metadata:
+            >>> reference_seqr = SeqRecord(Seq("ACGT"), id="reference", description="desc 1")
+            >>> seq1 = SeqRecord(Seq("ACGGT"), id="seq1", description="desc 2")
+            >>> seq2 = SeqRecord(Seq("AT"), id="seq2", description="desc 3")
+
+            To produce pairwise alignments:
+            >>> aligner = PairwiseAligner()
+            >>> pwa1 = next(aligner.align(reference_seqr, seq1))
+            >>> pwa2 = next(aligner.align(reference_seqr, seq2))
+
+            The msa retains the metadata from the original SeqRecord objects:
+            >>> msa = Alignment.from_alignments_with_same_reference([pwa1, pwa2])
+            >>> print(msa.format("fasta"))
+            >reference desc 1
+            ACG-T
+            >seq1 desc 2
+            ACGGT
+            >seq2 desc 3
+            A---T
+            <BLANKLINE>
+
+        """
+
+        if len(pwas) == 0:
+            raise ValueError("No pairwise alignments provided.")
+
+        # Validate that all pairwise alignments share the same reference
+        first_seqs = [pwa.sequences[0] for pwa in pwas]
+        # Same length (all types of references)
+        if not all(len(first_seq) == len(first_seqs[0]) for first_seq in first_seqs):
+            raise ValueError("All reference sequences must have the same length.")
+
+        # Same sequence (defined sequences only)
+        string_first_seqs = []
+        for first_seq in first_seqs:
+            # Exclude undefined sequences
+            if isinstance(first_seq, Seq) and isinstance(
+                first_seq._data, _UndefinedSequenceData
+            ):
+                continue
+            elif isinstance(first_seq, SeqRecord) and (
+                first_seq.seq is None
+                or isinstance(first_seq.seq._data, _UndefinedSequenceData)
+            ):
+                continue
+            elif isinstance(first_seq, SeqRecord):
+                string_first_seqs.append(str(first_seq.seq).upper())
+            else:
+                string_first_seqs.append(str(first_seq).upper())
```

Comment:
I would recommend:
```python
try:
    first_seq = first_seq.seq. # SeqRecord
except AttributeError:  # Seq or string
    pass
try:
    first_seq = str(first_seq)
except Bio.Seq.UndefinedSequenceError:
    continue
```
to be consistent with Python's duck-typing principle. Also, a `Seq` object may have a partially-defined sequence, so just testing for `_UndefinedSequenceData`  is not sufficient.